src/libraries/System.Runtime.InteropServices/src/System/Runtime/InteropServices/Marshalling/FreeThreadedStrategy.cs

```diff
@@ -12,6 +12,7 @@ internal sealed unsafe class FreeThreadedStrategy : IIUnknownStrategy
 
         void* IIUnknownStrategy.CreateInstancePointer(void* unknown)
         {
+            Marshal.AddRef((nint)unknown);
```

Comment:
The fact that we were missing this AddRef is how we didn't notice the leak. ComWrappers doesn't pass ownership in `CreateObject`, the implementor needs to `AddRef` if they create a wrapper. I forgot this and missed the ref ownership here. Since we weren't releasing the ref that we ended up owning from the marshaller, we didn't leak anything. Now that we're fixing the marshaller's ownership release, now we need to do the `AddRef` here to be correct.

Reply:
Can we assume the current testing caught this? Just curious if we have some basic lifetime reference testing.

Reply:
Yes, that's how I found it. The System.Transactions.Local tests crashed horribly on setup due to this issue.